File: java/src/main/java/com/meteroid/models/InvoicePaymentStatus.java
// This file is @generated
package com.meteroid.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.meteroid.Utils.ToQueryParam;

public enum InvoicePaymentStatus implements ToQueryParam {
    UNPAID("UNPAID"),
    PARTIALLY_PAID("PARTIALLY_PAID"),
    PAID("PAID"),
    ERRORED("ERRORED");
    private final String value;

    InvoicePaymentStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: python/meteroid/models/invoice_payment_status.py
# this file is @generated
import enum


class InvoicePaymentStatus(str, enum.Enum):
    UNPAID = "UNPAID"
    PARTIALLY_PAID = "PARTIALLY_PAID"
    PAID = "PAID"
    ERRORED = "ERRORED"
    PROCESSING = "PROCESSING"

    def __str__(self) -> str:
        return str(self.value)
